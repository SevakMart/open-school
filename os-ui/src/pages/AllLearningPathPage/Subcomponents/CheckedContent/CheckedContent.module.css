.checkedContentClass {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
}
.checkedContentClass img {
  margin-right: 12px;
}
.checkedContentClass p {
  display: inline;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  color: #5E617B;
}/*# sourceMappingURL=CheckedContent.module.css.map */