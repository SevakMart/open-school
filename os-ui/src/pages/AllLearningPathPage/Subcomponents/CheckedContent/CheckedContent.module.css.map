{"version":3,"sources":["CheckedContent.module.scss","../../../../styles/Mixins/FlexMixins.scss","CheckedContent.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss"],"names":[],"mappings":"AAKA;ECJE,aAAA;EACA,mBDIoB;ECHpB,mBDGyB;ECFzB,qBDEgC;AEDlC;AFGG;EACC,kBAAA;AEDJ;AFGG;EACC,eAAA;EGXF,yBCDY;EDEZ,kBHWmC;EGVnC,gBHU0C;EGT1C,eHS8C;EGR9C,iBHQmD;EACjD,cKZc;AHelB","file":"CheckedContent.module.css"}