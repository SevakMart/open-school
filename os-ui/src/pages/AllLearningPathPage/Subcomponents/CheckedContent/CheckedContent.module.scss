@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.checkedContentClass {
    @include flexStyle(row,flex-start,space-between);
    width:40%;
    label {
        @include fontStyle('Montserrat',normal,400,0.875rem,1.063rem);
        color:#5E617B;
    };
    input {
        border: 3px solid #F7F7FC;
    }
}

