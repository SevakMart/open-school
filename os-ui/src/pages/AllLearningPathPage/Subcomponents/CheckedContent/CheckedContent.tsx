import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate, useLocation } from 'react-router-dom';
import { addFilterParams, removeFilterParams } from '../../../../redux/Slices/AllLearningPathFilterParamsSlice';
import styles from './CheckedContent.module.scss';

const CheckedContent = ({ id, checkedContent, filterFeature }:
  {id:string, checkedContent:string, filterFeature:string}) => {
  const navigate = useNavigate();
  const location = useLocation();
  const dispatch = useDispatch();
  const [isChecked, setIsChecked] = useState(false);
  const params = new URLSearchParams(location.search);
  const { checkedContentClass } = styles;

  const handleChange = () => {
    setIsChecked((prevState) => !prevState);
  };
  useEffect(() => {
    if (params.has(checkedContent)) setIsChecked(true);
    /* dispatch(addFilterParams({ [filterFeature]: id })); */
  }, []);

  useEffect(() => {
    if (isChecked) {
      params.set(checkedContent, id);
      dispatch(addFilterParams({ [filterFeature]: id }));
      navigate(`/exploreLearningPaths?${params}`);
    } else {
      params.delete(checkedContent);
      dispatch(removeFilterParams({ [filterFeature]: id }));
      navigate(`/exploreLearningPaths?${params}`);
    }
  }, [isChecked]);

  return (
    <div className={checkedContentClass}>
      <input type="checkbox" name="checkbox" checked={!!params.has(checkedContent)} onChange={handleChange} id={id} />
      <label htmlFor={id}>{checkedContent}</label>
    </div>
  );
};
export default CheckedContent;
