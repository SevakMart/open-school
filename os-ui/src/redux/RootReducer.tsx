import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
