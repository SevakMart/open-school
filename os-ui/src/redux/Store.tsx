import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
