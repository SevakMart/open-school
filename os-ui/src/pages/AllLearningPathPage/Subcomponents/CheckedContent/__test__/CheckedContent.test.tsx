import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import CheckedContent from '../CheckedContent';

const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));

describe('Create test cases for CheckedContent component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CheckedContent
          id="1"
          checkedContent="React"
          filterFeature="category"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
