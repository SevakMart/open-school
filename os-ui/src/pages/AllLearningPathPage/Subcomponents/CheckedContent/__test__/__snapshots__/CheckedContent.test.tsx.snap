// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for CheckedContent component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="checkedContentClass"
  >
    <input
      data-testid="React"
      id="1"
      name="checkbox"
      type="checkbox"
    />
    <label
      data-testid="label of React"
      for="1"
    >
      React
    </label>
  </div>
</DocumentFragment>
`;
